Source/WebGPU/WebGPU/BindGroup.mm

```diff
@@ -459,23 +458,23 @@ static MTLPixelFormat metalPixelFormat(CVPixelBufferRef pixelBuffer, size_t plan
             return BindGroup::createInvalid(*this);
 
         for (ShaderStage stage : stages) {
-            if (!bindGroupLayout.bindingContainsStage(entry.binding, stage))
+            auto index = bindGroupLayout.indexForBinding(entry.binding, stage);
+            if (index == BindGroupLayout::NotFound)
                 continue;
 
-            auto& index = bindingIndexForStage[stage];
             if (bufferIsPresent) {
                 id<MTLBuffer> buffer = WebGPU::fromAPI(entry.buffer).buffer();
                 if (entry.offset > buffer.length)
                     return BindGroup::createInvalid(*this);
 
-                [argumentEncoder[stage] setBuffer:buffer offset:entry.offset atIndex:index++];
+                [argumentEncoder[stage] setBuffer:buffer offset:entry.offset atIndex:index];
                 resources.append({ buffer, MTLResourceUsageRead, metalRenderStage(stage) });
             } else if (samplerIsPresent) {
                 id<MTLSamplerState> sampler = WebGPU::fromAPI(entry.sampler).samplerState();
-                [argumentEncoder[stage] setSamplerState:sampler atIndex:index++];
+                [argumentEncoder[stage] setSamplerState:sampler atIndex:index];
             } else if (textureViewIsPresent) {
                 id<MTLTexture> texture = WebGPU::fromAPI(entry.textureView).texture();
-                [argumentEncoder[stage] setTexture:texture atIndex:index++];
+                [argumentEncoder[stage] setTexture:texture atIndex:index];
                 resources.append({ texture, MTLResourceUsageRead, metalRenderStage(stage) });
             } else if (externalTextureIsPresent) {
```

Comment:
We assume the indices generated by the WGSL compiler are linearly increasing, which I think is a reasonable assumption.

Reply:
With that assumption I don't think we need anything more complex than an integer.

Reply:
Isn't that, like, not true though? When we want to pack argument buffers, the indices will not be increasing. It would totally be possible for the color conversion matrices to not be directly after the yuv textures, if there are other textures in the bind group and we want to pack all the textures together.

Reply:
I think it is true and I'm not sure why we would implement such packing.

What is the benefit to packing textures together?